zk-sdk/src/range_proof/generators.rs

```diff
@@ -0,0 +1,156 @@
+use {
+    crate::range_proof::errors::RangeProofGeneratorError,
+    curve25519_dalek::{
+        digest::{ExtendableOutput, Update, XofReader},
+        ristretto::RistrettoPoint,
+    },
+    sha3::{Sha3XofReader, Shake256},
+};
+
+#[cfg(not(target_os = "solana"))]
+const MAX_GENERATOR_LENGTH: usize = u32::MAX as usize;
+
+/// Generators for Pedersen vector commitments that are used for inner-product proofs.
+struct GeneratorsChain {
+    reader: Sha3XofReader,
+}
+
+impl GeneratorsChain {
+    /// Creates a chain of generators, determined by the hash of `label`.
+    fn new(label: &[u8]) -> Self {
+        let mut shake = Shake256::default();
+        shake.update(b"GeneratorsChain");
+        shake.update(label);
+
+        GeneratorsChain {
+            reader: shake.finalize_xof(),
+        }
+    }
+
+    /// Advances the reader n times, squeezing and discarding
+    /// the result.
+    fn fast_forward(mut self, n: usize) -> Self {
+        for _ in 0..n {
+            let mut buf = [0u8; 64];
+            self.reader.read(&mut buf);
+        }
+        self
+    }
+}
+
+impl Default for GeneratorsChain {
+    fn default() -> Self {
+        Self::new(&[])
+    }
+}
+
+impl Iterator for GeneratorsChain {
+    type Item = RistrettoPoint;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut uniform_bytes = [0u8; 64];
+        self.reader.read(&mut uniform_bytes);
+
+        Some(RistrettoPoint::from_uniform_bytes(&uniform_bytes))
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (usize::max_value(), None)
+    }
+}
+
+#[allow(non_snake_case)]
+#[derive(Clone)]
+pub struct BulletproofGens {
```

Comment:
I thought `Bulletproof` would get changed to `RangeProof`, is that not the case?

Reply:
Oh right, I forgot to mention that I will do so in a follow-up, but might be good to do this here since there aren't that many commits!